### PR TITLE
Custom domain smoke test

### LINKED ISSUE #1597: Smoke test for custom domain
Support for custom domain instead of hard-coded `cluster.local` landed in #1569. We should create a e2e smoke test for it too.

### PR DESCRIPTION
**Pull request by jnummelin**
## Description

E2E smoke test for #1569

Includes also a fix when parsing a partial `spec.network` yaml config.

Fixes #1597 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [x] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: inttest/customdomain/customdomain_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+	"github.com/k0sproject/k0s/inttest/common"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type CustomDomainSuite struct {
+	common.FootlooseSuite
+}
+
+func (s *CustomDomainSuite) TestK0sGetsUpWithCustomDomain() {
```
Comment (twz123) [inline inttest/customdomain/customdomain_test.go:32]: Would be cool if we could check if the custom domain actually gets picked up.

**Review by twz123 — APPROVED** [review]

File: inttest/customdomain/customdomain_test.go (hunk 1 above)
Comment (jnummelin) [reply inttest/customdomain/customdomain_test.go:32]: Any preferences how to do that properly?

Comment (twz123) [reply inttest/customdomain/customdomain_test.go:32]: Not sure. Probably there are more clever ways than this, but I think Inspecting a pod's /etc/resolv.conf could work.

Comment (jnummelin) [reply inttest/customdomain/customdomain_test.go:32]: @twz123 Added check for this in the last commit, PTAL

**Review by twz123 — APPROVED** [review]

**Pull request merged**